api/v1alpha1/flowcollector_types.go

```diff
@@ -46,6 +46,10 @@ type FlowCollectorSpec struct {
 	// Loki contains settings related to the loki client
 	Loki FlowCollectorLoki `json:"loki,omitempty"`
 
+	// Kafka configurations, if empty the operator will deploy a all-in-one FLP
+	// +optional
+	Kafka *FlowCollectorKafka `json:"kafka,omitempty"`
```

Comment:
Kafka here is ingress or egress ... we support both, and we need two configurations one when we ingress data from Kafka into FLP and one when we export from FLP data into kafka. 

It will be good to be more explicit so users will understand exactly what is going on

Reply:
I see that you use the term "single" for this simple case and I would recommend reversing the logic and using the term `distributed` for kafka and the distributed approach ... this is just my way of thinking on things, starting from simple == simple and moving into the more complicated distributed approach and not vice versa

Reply:
I do not remember if we have TLS support (or any other security) for Kafka ... but we need that support both in the operator and in the FLP code.

Reply:
@eranra about your different comments here:
- the kafka configuration block here is more global, this is the kafka configuration as for kafka in flow collection, this block will also concern ebpf agent once we will implement kafka support in the agent
- FLPReconciler and singleFLPReconciler do not oppose,  the model here is to have one parent reconciler which create three reconciler, one for each possible FLP configuration (all-in-one, ingestor and transformer) both reconciler types are used whatever the configuration is. I would prefer to keep the FLPReconciler name for the parent, since this is the one that is getting exposed, are you fine if I rename singleFLPReconciler to singleDeploymentReconciler for more clarity (we can remove the FLP part since this is an internal object and we are already in the FLP package)?
- I did not dig much into TLS yet, I can create a dedicated task about kafka security, would you be fine with that ?

Reply:
@OlivierCazade with regards to the configuration ... I agree this fits, I think that I learned more while reading the code, so this is totally ok. 

singleDeploymentReconciler -- it is good --- agree this will help to make things more understood :-) 

TLS -- yes we might need that for production